src/components/show-products/index.js

```diff
@@ -43,27 +67,46 @@ export const ShowProducts = () => {
 
   return (
     <ContentContainer>
+      <form className="filter-form">
+        <label htmlFor="filter">
+          Filter items
+          <input
+            className="input-item"
+            type="search"
+            id="filter"
+            name="filter"
+            aria-label="Search through list content"
+            placeholder="Start typing here..."
+            value={searchTerm}
+            onChange={(event) => setSearchTerm(event.target.value)}
+          />
+        </label>
+      </form>
       <ul>
-        {products?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
-          <li className="checkbox-item" key={id}>
-            <label htmlFor={id} className="checkbox-label">
-              <input
-                type="checkbox"
-                id={id}
-                name={productName}
-                onChange={() => handleOnChange(id)}
-                checked={
-                  lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
-                }
-                ariaLabel={timeFrames[timeFrame]}
-              />
-              <span
-                className={`checkmark checkbox-timeFrame-${timeFrame}`}
-              ></span>
-              <span className="checkbox-name">{productName}</span>
-            </label>
-          </li>
-        ))}
+        {list.length > 0 ? (
+          list.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
+            <li className="checkbox-item" key={id}>
+              <label htmlFor={id} className="checkbox-label">
+                <input
+                  type="checkbox"
+                  id={id}
+                  name={productName}
+                  onChange={() => handleCheckboxChange(id)}
+                  checked={
+                    lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
+                  }
+                  aria-label={TimeFrames[timeFrame]}
```

Comment:
This `aria-label` is so magic and dynamic. 

**_Only an idea_**, change this object to `TimeLabels` or `FrequencyLabels` as names 😀 .

Reply:
I think it's a good idea to make the names more meaningful 😸

Reply:
I think its a good idea, much more clear to what it describes :P

Reply:
@IsabellaSoares @susird  feel free to change this, girls. I think it's a good idea, but don't forget to change all matches in TimeLabels throught the files